File: rent-now-backend/src/main/java/pl/holowinska/rentnowbackend/model/rq/AccommodationRQ.java
package pl.holowinska.rentnowbackend.model.rq;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.*;
import pl.holowinska.rentnowbackend.model.enums.ConvenienceType;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class AccommodationRQ {

    @NotNull
    private String name;
    @NotNull
    private AddressRQ addressRQ;
    @NotNull
    private UUID userUUID;
    @NotNull
    @Positive
    private BigDecimal priceForDay;
    private BigDecimal squareFootage;
    @NotBlank
    private String description;
    private HashMap<ConvenienceType, BigDecimal> conveniences;
    @NotNull
    private Integer maxNoOfPeople;
}

File: rent-now-ui/src/app/data/model/rq/AccommodationRQ.ts
import {AddressRQ} from "./AddressRQ";
import {ConvenienceType} from "../common/ConvenienceType";

export interface AccommodationRQ {
  addressRQ: AddressRQ;
  userUUID: string;
  name: string;
  priceForDay: number;
  squareFootage: number;
  description: string;
  conveniences: Map<ConvenienceType, number>;
  maxNoOfPeople: number;
}
